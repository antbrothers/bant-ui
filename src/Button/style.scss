.bg-light-purple {
    // background-color: aquamarine;
}