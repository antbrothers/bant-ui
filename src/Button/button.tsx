import React from 'react';
import './style.scss'
interface IProps {
  /**
   * Text for the button
   */
  text: string;
}

export const Button = (props: IProps) => {
  return (
    <div className='f6 link dim br3 ph3 pv2 mb2 dib white bg-light-purple'>
      {props.text}
    </div>
  )
}; 