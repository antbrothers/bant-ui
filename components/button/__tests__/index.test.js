import React from "react";
import assert from "power-assert";
import { render, shallow } from "enzyme";
import toJson from "enzyme-to-json";
import Button from "../index";
// import { LoadingIcon } from "../../icon";

describe("<Button/>", () => {
  it("should render a <Button/> components", () => {
    const wrapper = render(
      <div>
        <Button>你好</Button>
        <Button type="primary">你好</Button>
        <Button type="info">你好</Button>
        <Button type="warning">你好</Button>
        <Button type="error">你好</Button>
        <Button type="success">你好</Button>
        <Button disabled>你好</Button>
        <Button loading>加载中..</Button>
        <Button block>100%</Button>
        <Button circle>圆形</Button>
        <Button type="primary" size="large">
          大
        </Button>

        <Button type="info">中</Button>

        <Button type="warning" size="small">
          小
        </Button>
      </div>
    );
    expect(toJson(wrapper)).toMatchSnapshot();
  });

  // it("should find cuke-button classnames", () => {
  //   const wrapper = render(
  //     <div>
  //       <Button>你好</Button>
  //       <Button type="primary">你好</Button>
  //       <Button type="info">你好</Button>
  //       <Button type="warning">你好</Button>
  //       <Button type="error">你好</Button>
  //       <Button type="success">你好</Button>
  //       <Button disabled>你好</Button>
  //       <Button loading>加载中..</Button>
  //       <Button block>100%</Button>
  //       <Button circle>100%</Button>
  //     </div>
  //   );
  //   assert(wrapper.find(".bant-button").length >= 1);
  //   assert(wrapper.find(".bant-button-primary").length === 1);
  //   assert(wrapper.find(".bant-button-info").length === 1);
  //   assert(wrapper.find(".bant-button-warning").length === 1);
  //   assert(wrapper.find(".bant-button-error").length === 1);
  //   assert(wrapper.find(".bant-button-success").length === 1);
  //   assert(wrapper.find(".bant-button-disabled").length === 1);
  //   assert(wrapper.find(".bant-button-loading").length === 1);
  //   assert(wrapper.find(".bant-button-block").length === 1);
  //   assert(wrapper.find(".bant-button-circle").length === 1);
  // });

  // it("should can trigger click event", () => {
  //   const onClick = jest.fn();
  //   const wrapper = shallow(<Button onClick={onClick}>黄瓜ui</Button>);
  //   wrapper.find("button").simulate("click");
  //   expect(onClick).toHaveBeenCalled();
  // });
  // it("should can not trigger click event", () => {
  //   const onClick = jest.fn();
  //   const wrapper = shallow(
  //     <Button onClick={onClick} disabled>
  //       黄瓜ui
  //     </Button>
  //   );
  //   wrapper.find("button").simulate("click");
  //   expect(onClick).not.toHaveBeenCalled();
  // });
  // it("should can not render loading icon when mode is circle", () => {
  //   const wrapper = shallow(
  //     <Button loading circle>
  //       黄瓜ui
  //     </Button>
  //   );
  //   // assert(wrapper.find(LoadingIcon).length === 0);
  //   assert(wrapper.find(".cuke-loading").length === 0);
  // });

  // it("should render link", () => {
  //   const wrapper = shallow(<Button href="/test">黄瓜ui</Button>);
  //   assert(wrapper.find(".bant-button-link").length >= 1);
  // });

  // it("should can not click link when disabled", () => {
  //   const onClick = jest.fn();
  //   const wrapper = shallow(
  //     <Button href="/test" onClick={onClick} disabled>
  //       黄瓜ui
  //     </Button>
  //   );
  //   wrapper.find("a").simulate("click");
  //   expect(onClick).not.toHaveBeenCalled();
  // });
});
