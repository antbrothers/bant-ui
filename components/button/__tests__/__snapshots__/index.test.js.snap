// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Button/> should render a <Button/> components 1`] = `
<div>
  <button
    class="bant-button bant-button-default bant-button-normal"
    type="button"
  >
    <span>
      你好
    </span>
  </button>
  <button
    class="bant-button bant-button-primary"
    type="button"
  >
    <span>
      你好
    </span>
  </button>
  <button
    class="bant-button bant-button-info"
    type="button"
  >
    <span>
      你好
    </span>
  </button>
  <button
    class="bant-button bant-button-warning"
    type="button"
  >
    <span>
      你好
    </span>
  </button>
  <button
    class="bant-button bant-button-error"
    type="button"
  >
    <span>
      你好
    </span>
  </button>
  <button
    class="bant-button bant-button-success"
    type="button"
  >
    <span>
      你好
    </span>
  </button>
  <button
    class="bant-button bant-button-normal bant-button-disabled"
    disabled=""
    type="button"
  >
    <span>
      你好
    </span>
  </button>
  <button
    class="bant-button bant-button-default bant-button-normal bant-button-loading"
    disabled=""
    type="button"
  >
    <span>
      加载中..
    </span>
  </button>
  <button
    class="bant-button bant-button-default bant-button-normal bant-button-block"
    type="button"
  >
    <span>
      100%
    </span>
  </button>
  <button
    class="bant-button bant-button-default bant-button-normal bant-button-circle"
    type="button"
  >
    <span>
      圆形
    </span>
  </button>
  <button
    class="bant-button bant-button-primary bant-button-size-large"
    type="button"
  >
    <span>
      大
    </span>
  </button>
  <button
    class="bant-button bant-button-info"
    type="button"
  >
    <span>
      中
    </span>
  </button>
  <button
    class="bant-button bant-button-warning bant-button-size-small"
    type="button"
  >
    <span>
      小
    </span>
  </button>
</div>
`;
